platform/posix/transport/src/plaintext_posix.c

```diff
@@ -113,30 +69,70 @@ int32_t Plaintext_Recv( const NetworkContext_t * pNetworkContext,
                         void * pBuffer,
                         size_t bytesToRecv )
 {
-    int32_t bytesReceived = 0;
+    int32_t bytesReceived = -1, selectStatus = -1, getTimeoutStatus = -1;
+    struct timeval recvTimeout;
+    socklen_t recvTimeoutLen;
+    fd_set readfds;
+
+    assert( pNetworkContext != NULL );
+    assert( pBuffer != NULL );
+    assert( bytesToRecv > 0 );
+
+    /* Get receive timeout from the socket to use as the timeout for #select. */
+    recvTimeoutLen = ( socklen_t ) sizeof( recvTimeout );
+    getTimeoutStatus = getsockopt( pNetworkContext->socketDescriptor,
+                                   SOL_SOCKET,
+                                   SO_RCVTIMEO,
+                                   &recvTimeout,
+                                   &recvTimeoutLen );
+
+    /* Make #select return immediately if getting the timeout failed. */
+    if( getTimeoutStatus < 0 )
+    {
+        recvTimeout.tv_sec = 0;
+        recvTimeout.tv_usec = 0;
+    }
 
-    bytesReceived = ( int32_t ) recv( pNetworkContext->socketDescriptor, pBuffer, bytesToRecv, 0 );
+    FD_ZERO( &readfds );
+    FD_SET( pNetworkContext->socketDescriptor, &readfds );
+    /* Check if there is data to read from the socket. */
+    selectStatus = select( pNetworkContext->socketDescriptor + 1,
+                           &readfds,
+                           NULL,
+                           NULL,
+                           &recvTimeout );
+
+    if( selectStatus > 0 )
+    {
+        /* The socket is available for receiving data. */
+        bytesReceived = ( int32_t ) recv( pNetworkContext->socketDescriptor,
+                                          pBuffer,
+                                          bytesToRecv,
+                                          0 );
+    }
+    else if( selectStatus < 0 )
+    {
+        /* An error occurred while polling. */
+        bytesReceived = -1;
```

Comment:
I'm concerned that with this change, an error return value will purely depend on the call to `select` instead of the `recv` call. 
Are there some error return values of `recv` that will be ignored with this approach?

Looking at the Linux man page of [`select`](https://man7.org/linux/man-pages/man2/select.2.html), I am hoping that the `EBADF` return value could catch errors encountered with _open_ file descriptors.
![image](https://user-images.githubusercontent.com/5158611/92281470-87e7ed80-eeb0-11ea-8f29-281935eab5cb.png)